lumeer-storage/lumeer-storage-api/src/main/java/io/lumeer/storage/api/query/SearchQuery.java

```diff
@@ -18,73 +18,26 @@
  */
 package io.lumeer.storage.api.query;
 
-import io.lumeer.storage.api.filter.AttributeFilter;
+import io.lumeer.api.model.QueryStem;
 
-import java.util.Collections;
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
 public class SearchQuery extends DatabaseQuery {
 
-   private final String fulltext;
-   private final Set<String> collectionIds;
-   private final Set<String> linkTypeIds;
-   private final Set<String> documentIds;
-   private final Set<AttributeFilter> filters;
+   private final List<SearchQueryStem> stems;
 
    private SearchQuery(Builder builder) {
       super(builder);
 
-      this.fulltext = builder.fulltext;
-      this.collectionIds = builder.collectionIds;
-      this.linkTypeIds = builder.linkTypeIds;
-      this.documentIds = builder.documentIds;
-      this.filters = builder.filters;
+      this.stems = builder.stems;
    }
 
-   public String getFulltext() {
-      return fulltext;
-   }
-
-   public Set<String> getLinkTypeIds() {
-      return linkTypeIds != null ? Collections.unmodifiableSet(linkTypeIds) : Collections.emptySet();
-   }
-
-   public Set<String> getDocumentIds() {
-      return documentIds != null ? Collections.unmodifiableSet(documentIds) : Collections.emptySet();
-   }
-
-   public Set<String> getCollectionIds() {
-      return collectionIds != null ? Collections.unmodifiableSet(collectionIds) : Collections.emptySet();
-   }
-
-   public Set<AttributeFilter> getFilters() {
-      return filters != null ? Collections.unmodifiableSet(filters) : Collections.emptySet();
-   }
-
-   public boolean isFulltextQuery() {
-      return fulltext != null && !fulltext.isEmpty();
-   }
-
-   public boolean isCollectionIdsQuery() {
-      return collectionIds != null && !collectionIds.isEmpty();
-   }
-
-   public boolean isLinkTypeIdsQuery() {
-      return linkTypeIds != null && !linkTypeIds.isEmpty();
-   }
-
-   public boolean isDocumentIdsQuery() {
-      return documentIds != null && !documentIds.isEmpty();
-   }
-
-   public boolean isFiltersQuery() {
-      return filters != null && !filters.isEmpty();
-   }
-
-   public boolean isBasicQuery() {
-      return !isFulltextQuery()  && !isLinkTypeIdsQuery() && !isDocumentIdsQuery() && !isCollectionIdsQuery();
+   public List<SearchQueryStem> getStems() {
+      return stems;
```

Comment:
Since this class is annotated as `@Immutable`, we should return immutable list here:
```java
return Collections.unmodifiableList(stems);
```